atomic_reactor/core.py

```diff
@@ -272,10 +277,60 @@ def __init__(self, base_url=None, retry_times=DOCKER_MAX_RETRIES,
 
         if hasattr(docker, 'AutoVersionClient'):
             client_kwargs['version'] = 'auto'
-        client_kwargs['retry'] = retry_times
+        self.retry_times = retry_times
+        client_kwargs['retry'] = self.retry_times
 
         self.d = WrappedDocker(**client_kwargs)
 
+    def retry_generator(self, fnc_name, function, *args, **kwargs):
+        retry_times = int(kwargs.pop('retry_times', self.retry_times))
+        retry_delay = DOCKER_BACKOFF_FACTOR
+        retry_client_statuses = DOCKER_CLIENT_STATUS_RETRY
+
+        for counter in range(retry_times + 1):
+            exc = None
+            try:
+                logs_gen = function(*args, **kwargs)
+                cmd_result = atomic_reactor.util.wait_for_command(logs_gen)
+
+            except ProtocolError as e:
+                if counter != retry_times:
+                    exc = e
+                else:
+                    if hasattr(exc, 'message'):
+                        raise RetryGeneratorException("Failed to %s image %s: %s" %
+                                                      (fnc_name, args, e.message))
+                    else:
+                        raise RetryGeneratorException("Failed to %s image %s: %s" %
+                                                      (fnc_name, args, e.with_traceback))
+
+            except APIError as e:
+                if (e.response.status_code in retry_client_statuses and counter != retry_times):
+                    exc = e
+                else:
+                    raise RetryGeneratorException("Failed to %s image %s: %s" %
+                                                  (fnc_name, args, e.response.content))
+
+            if exc:
+                if (isinstance(exc, ProtocolError)):
+                    if hasattr(exc, 'message'):
+                        logger.info("retrying %s - %s on %s", fnc_name, args, exc.message)
```

Comment:
`Exception.message` is deprecated

Reply:
python2 has message, python3 has that traceback

Reply:
That's how deprecation works - its still there, but no point to be used.

If we really want a traceback (I doubt that it would be useful) `traceback` module should be used instead

Reply:
ahh ok will use .args